java/java.source.base/src/org/netbeans/modules/java/source/save/CasualDiff.java

```diff
@@ -535,8 +535,18 @@ public int endPos(JCTree t) {
             VariableTree vt = fgt.getVariables().get(fgt.getVariables().size() - 1);
             return TreeInfo.getEndPos((JCTree)vt, oldTopLevel.endPositions);
         }
-        return TreeInfo.getEndPos(t, oldTopLevel.endPositions);
+        int endPos = TreeInfo.getEndPos(t, oldTopLevel.endPositions);
+
+        // [NETBEANS-4299], catch a synthetic annotation attribute and try rhs
+        if (endPos == Position.NOPOS && t instanceof JCAssign) {
+            JCAssign assign = (JCAssign)t;
+            if (assign.lhs instanceof JCIdent
+                    && ((JCIdent)assign.lhs).sym instanceof Symbol.MethodSymbol) {
```

Comment:
I think the "&& ... instanceof MethodSymbol" is probably unnecessary.

Reply:
I guess a regular/real assignment would have been in oldTopLevel.endPositions. So since it failed to get endPos as an assignment, all that's needed is to simply check the rhs (the end should be after that). Checking the lhs shouldn't be needed at all. I was being most conservative, but any rhs match should be good. I guess it could just be
```
// [NETBEANS-4299], catch a synthetic annotation attribute and try rhs
if (endPos == Position.NOPOS && t instanceof JCAssign) {
    endPos = TreeInfo.getEndPos(((JCAssign)t).rhs, oldTopLevel.endPositions);
}
```
That feels less conservative and I don't know anything about the quirks in the trees.

@jlahoda  your call. Do you want either change or as is?

Reply:
I'd say - simplify the code. (We will see if some tests fail.) Overall, touching symbols too much inside CausalDiff is not very good, although in this particular case it probably does not matter much.

Reply:
Simplified. Since first check is NOPOS, probably doesn't matter how much stuff is done; but I like simple. My manual tests passed.

Do you believe this is safe for 12.0? There's that milestone thingy... And/or might add @ebarboni as reviewer.